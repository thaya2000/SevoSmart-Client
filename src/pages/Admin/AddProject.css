/* AddProject.css */

/* General styles */
.admin-panel {
    display: flex;
    flex-direction: row;
  }
  
  .admin-sidebar {
    background-color: #2d3748;
    width: 16rem; /* w-64 */
    padding: 1rem; /* p-4 */
    color: white;
  }
  
  .admin-content {
    padding: 2rem; /* p-8 */
    background-color: white;
    flex: 1;
  }
  
  .admin-tableh {
    font-size: 2rem; /* text-4xl */
    font-weight: 500; /* font-medium */
    margin-bottom: 2rem; /* mb-8 */
    color: #152f64; /* text-blue-900 */
    text-align: center; /* flex justify-center */
  }
  
  .mb-6 {
    margin-bottom: 1.5rem;
  }
  
  .shadow {
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
  }
  
  .block {
    display: block;
  }
  
  .text-blue-900 {
    color: #1a202c;
  }
  
  .text-m {
    font-size: 1rem;
  }
  
  .font-bold {
    font-weight: bold;
  }
  
  .py-2 {
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
  }
  
  .px-3 {
    padding-left: 0.75rem;
    padding-right: 0.75rem;
  }
  
  .md\\:mb-0 {
    margin-bottom: 0;
  }
  
  
  
  .mt-1 {
    margin-top: 0.25rem;
  }
  
  .w-full {
    width: 100%;
  }
  
  .max-w-sm {
    max-width: 20rem;
  }
  
  .text-gray-700 {
    color: #4a5568;
  }
  
  .leading-tight {
    line-height: 1.25;
  }
  
  .focus\\:outline-none {
    outline: 2px solid transparent;
    outline-offset: 2px;
  }
  
  .focus\\:shadow-outline {
    box-shadow: 0 0 0 3px rgba(66, 153, 225, 0.5);
  }
  
  .my-8 {
    margin-top: 2rem;
    margin-bottom: 2rem;
  }
  
  .bg-red-700 {
    background-color: #c53030;
  }
  
  .hover\\:bg-grey-700:hover {
    background-color: #4a5568;
  }
  
  .text-white {
    color: white;
  }
  
  .rounded {
    border-radius: 0.25rem;
  }
  
  .space-y-4 > * + * {
    margin-top: 1rem;
  }
  
 
  @media (max-width: 768px) {
    .admin-panel {
      flex-direction: column;
    }
  
    .admin-tableh {
      font-size: 1.5rem; 
    }
  
    .mb-6 {
      margin-bottom: 1rem; 
    }
  
    .text-m {
      font-size: 0.875rem; 
    }
  
    .form-label {
      width: 100%; 
    }
  
    .form-input {
      width: 100%; 
      max-width: 100%; 
      padding: 0.375rem 0.5rem; 
    }
  }
  