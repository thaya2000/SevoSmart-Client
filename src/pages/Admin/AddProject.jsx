import React, { useState, useEffect } from 'react';
import { useNavigate } from "react-router-dom";
import toast from "react-hot-toast";
import axios from "axios";

const AddProject = () => {
    const [name, setName] = useState("");
    const [description, setDescription] = useState("");
    const [images, setImages] = useState([]);
    const [imagePreviews, setImagePreviews] = useState([]);

    const navigate = useNavigate();

    useEffect(() => {
        console.log("Name:", name);
        console.log("Description:", description);
        console.log("images:", images);
    }, [name, description, images]);

    const handleSubmit = async (e) => {
        e.preventDefault();
        try {
            const PastProjectData = new FormData();
            PastProjectData.append("name", name);
            PastProjectData.append("description", description);
            for (let i = 0; i < images.length; i++) {
                PastProjectData.append("images", images[i]); // Append each selected image file
            }

            const { data } = await axios.post(
                "http://localhost:8080/api/v1/admin/pastproject",
                PastProjectData
            );
            console.log(data);
            if (data?.error) {
              toast.error(data.error);
            } else {
              toast.success("Project is uploaded successfully");
              navigate("/past-projects");
            }
        } catch (err) {
            console.log(err);
      toast.error("Project upload failed. Try again.");
        }
    };

    const handleImageChange = (e) => {
        const selectedImages = e.target.files;
        const imagePreviews = [];
        const newImages = [];

        // Loop through each selected image
        for (let i = 0; i < selectedImages.length; i++) {
            const reader = new FileReader();
            const selectedImage = selectedImages[i];

            // Read each image and push its data URL to the imagePreviews array
            reader.onload = () => {
                imagePreviews.push(reader.result);
                // If all images have been read, update the state with the array of previews
                if (imagePreviews.length === selectedImages.length) {
                    setImagePreviews(imagePreviews);
                }
            };

            reader.readAsDataURL(selectedImage);
            newImages.push(selectedImage); // Push the selected image to the newImages array
        }
        setImages(newImages); // Update the images state with the new array of images
    };

    return (
        <div className="p-8 bg-indigo-950">
            <h1 className="text-6xl font-medium mb-4 text-white">Add Project</h1>
            <form onSubmit={handleSubmit}>
                <div className="flex flex-col mb-4">
                    <label className="text-white mb-2">Project Name</label>
                    <input type="text" value={name} onChange={(e) => setName(e.target.value)} className="bg-white rounded-lg px-4 py-2" />
                </div>

                <div className="flex flex-col mb-4">
                    <label className="text-white mb-2">Project Description</label>
                    <textarea value={description} onChange={(e) => setDescription(e.target.value)} className="bg-white rounded-lg px-4 py-2" />
                </div>

                <div className="flex flex-col mb-4">
                    <label className="text-white mb-2">Project Images</label>
                    <input type="file" accept="image/*" multiple onChange={handleImageChange} className="mb-2" />
                </div>
                <div className="mb-4 flex flex-wrap justify-center">
                    {imagePreviews && imagePreviews.map((preview, index) => (
                        <img key={index} src={preview} alt={`Product Preview ${index + 1}`} className="w-1/5 mx-2 mb-2" />
                    ))}
                </div>

                <button className="bg-blue-700 hover:bg-blue-800 text-white font-medium rounded-lg text-sm px-5 py-2.5 focus:outline-none">
                    Add Project
                </button>
            </form>
        </div>
    );
};

export default AddProject;
