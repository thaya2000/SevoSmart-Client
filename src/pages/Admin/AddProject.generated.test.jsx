import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import AddProject from './AddProject';

// Mocking necessary modules
vi.mock("react-router-dom", () => ({
  useNavigate: vi.fn()  // Mocking the navigate function
}));
vi.mock("react-hot-toast", () => ({
  toast: {
    success: vi.fn(),
    error: vi.fn()
  }
}));
vi.mock("axios");

describe('<AddProject>', () => {
  it('should render component', () => {
    const { asFragment } = render(<AddProject />);
    expect(asFragment()).toMatchSnapshot();
  });
});
