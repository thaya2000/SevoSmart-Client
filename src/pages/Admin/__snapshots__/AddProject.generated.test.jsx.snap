// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AddProject> > should render component 1`] = `
<DocumentFragment>
  <div
    class="p-8 bg-indigo-950"
  >
    <h1
      class="text-6xl font-medium mb-4 text-white"
    >
      Add Project
    </h1>
    <form>
      <div
        class="flex flex-col mb-4"
      >
        <label
          class="text-white mb-2"
        >
          Project Name
        </label>
        <input
          class="bg-white rounded-lg px-4 py-2"
          type="text"
          value=""
        />
      </div>
      <div
        class="flex flex-col mb-4"
      >
        <label
          class="text-white mb-2"
        >
          Project Description
        </label>
        <textarea
          class="bg-white rounded-lg px-4 py-2"
        />
      </div>
      <div
        class="flex flex-col mb-4"
      >
        <label
          class="text-white mb-2"
        >
          Project Images
        </label>
        <input
          accept="image/*"
          class="mb-2"
          multiple=""
          type="file"
        />
      </div>
      <div
        class="mb-4 flex flex-wrap justify-center"
      />
      <button
        class="bg-blue-700 hover:bg-blue-800 text-white font-medium rounded-lg text-sm px-5 py-2.5 focus:outline-none"
      >
        Add Project
      </button>
    </form>
  </div>
</DocumentFragment>
`;
